plugins/kubernetes/lib/samson_kubernetes/samson_plugin.rb

```diff
@@ -63,3 +63,18 @@ def self.retry_on_connection_errors(&block)
 end
 
 Samson::Hooks.callback(:deploy_group_includes) { :kubernetes_cluster }
+
+Samson::Hooks.callback(:stage_clone) do |old_stage, new_stage|
+  roles_to_copy = Kubernetes::DeployGroupRole.where(
+    project: old_stage.project,
+    deploy_group: old_stage.deploy_groups.last
+  )
+
+  roles_to_copy.each do |dgr|
```

Comment:
clone does not always have to have 1 deploy-group, so should be `new_stage.deploy_groups.each do`